src/headers/frame_header.rs

```diff
@@ -0,0 +1,494 @@
+            ],
+            // TODO(Tomáš Král): It would be nice to use struct update syntax like this:
+            // FrameHeader {
+            //     fields that we actually care about
+            //     ..FrameHeader::default_from_nonserialized(&nonserialized)
+            // }
+            FrameHeader {
+                all_default: false,
+                frame_type: FrameType::RegularFrame,
+                encoding: Encoding::VarDCT,
+                flags: 0,
+                do_ycbcr: false,
+                jpeg_upsampling: [0, 0, 0],
+                upsampling: 1,
+                ec_upsampling: vec![],
+                group_size_shift: 1,
+                x_qm_scale: 2,
+                b_qm_scale: 2,
+                passes: Passes::default(),
+                lf_level: 0,
+                have_crop: false,
+                x0: 0,
+                y0: 0,
+                width: 0,
+                height: 0,
+                blending_info: BlendingInfo::default(),
+                ec_blending_info: vec![],
+                duration: 0,
+                timecode: 0,
+                is_last: true,
+                save_as_reference: 0,
+                save_before_ct: false,
+                name: String::from(""),
+                restoration_filter: RestorationFilter::default(),
+                extensions: Extensions::default(),
+            }
+        );
+    }
+
+    #[test]
+    fn test_extra_channel() {
+        test_frame_header!(
+            vec![
+                0xFF, 0x0A, 0x41, 0xC0, 0x4A, 0x08, 0x10, 0x10, 0x00, 0xE4, 0x01, 0x4B, 0x28, 0x36,
+                0x56, 0x1F, 0xDC, 0x4B, 0x28, 0x98, 0x10, 0x01, 0x55, 0x21, 0xC4, 0x30, 0x06, 0x50,
+                0x87, 0x61, 0xAB, 0x2A, 0xB2, 0x17, 0x03, 0x02, 0xA0, 0x97, 0xCC, 0x08, 0x00, 0xC3,
+                0x63, 0x80, 0x49, 0x66, 0x12, 0x04, 0x78, 0x2C, 0xD6, 0x89, 0x53, 0xEF, 0xF9, 0x15,
+                0xFC, 0xD1, 0x6B, 0xC4, 0xF3, 0xC0, 0x0E, 0xA9, 0x8D, 0xB6, 0x16, 0x4E, 0x5C, 0x70,
+                0x06, 0xE2, 0x07, 0x12, 0x62, 0xEC, 0x6C, 0xBE, 0x7C, 0x16, 0xDC, 0x72, 0xCE, 0xF3,
+                0xC1, 0xA2, 0xE2, 0x0A, 0xC8, 0xF9, 0xA1, 0x8C, 0xDA, 0xCF, 0xE3, 0xE8, 0x27, 0xDA,
+                0x66, 0xE2, 0xD6, 0x20, 0x2A, 0x38, 0xC1, 0xF7, 0xD0, 0x66, 0xED, 0xD2, 0xE0, 0x04,
+                0x42, 0x3A, 0x2A, 0x99, 0x2C, 0x12, 0x19, 0x9D, 0x9E, 0x83, 0x28, 0x54, 0x81, 0x55,
+                0x83, 0x3D, 0x69, 0x00, 0x1D, 0x03
+            ],
+            FrameHeader {
+                all_default: false,
+                frame_type: FrameType::RegularFrame,
+                encoding: Encoding::Modular,
+                flags: 0,
+                do_ycbcr: false,
+                jpeg_upsampling: [0, 0, 0],
+                upsampling: 1,
+                ec_upsampling: vec![1],
+                group_size_shift: 2,
+                // libjxl x_qm_scale = 2, but condition is false (should be 3 according to the draft)
```

Comment:
@jonsneyers not that it matters, but we should probably also figure this one out or fix the spec.